imagepipeline/src/main/java/com/facebook/imagepipeline/producers/ResizeAndRotateProducer.java

```diff
@@ -237,19 +256,21 @@ private static TriState shouldTransform(
     final float heightRatio = ((float) resizeOptions.height) / height;
     float ratio = Math.max(widthRatio, heightRatio);
 
-    // TODO: The limit is larger than this on newer devices. The problem is to get the real limit,
-    // you have to call Canvas.getMaximumBitmapWidth/Height on a real HW-accelerated Canvas.
-    if (width * ratio > BitmapUtil.MAX_BITMAP_SIZE) {
-      ratio = BitmapUtil.MAX_BITMAP_SIZE / width;
-    }
-    if (height * ratio > BitmapUtil.MAX_BITMAP_SIZE) {
-      ratio = BitmapUtil.MAX_BITMAP_SIZE / height;
-    }
+    // TODO: Both width and height should be equal or smaller than device's max bitmap size
+    float maxBitmapSize = Math.max(resizeOptions.width, resizeOptions.height);
+
+    if (width * ratio > maxBitmapSize) {
+	      ratio = maxBitmapSize / width;
+	    }
+	    if (height * ratio > maxBitmapSize) {
+	      ratio = maxBitmapSize / height;
+	    }
+    FLog.v(TAG, "Resize ratio: %f", ratio);
     return ratio;
   }
 
   @VisibleForTesting static int roundNumerator(float maxRatio) {
-    return (int) (ROUNDUP_FRACTION + maxRatio * JpegTranscoder.SCALE_DENOMINATOR);
+    return (int) (maxRatio * JpegTranscoder.SCALE_DENOMINATOR);
   }
 
   private static int getScaleNumerator(
```

Comment:
We should add clamping in this method, right after `int numerator = roundNumerator(ratio)`.

```
int numerator = roundNumerator(ratio);
while (widthAfterRotation * numerator / JpegTranscoder.SCALE_DENOMINATOR > MAX_WIDTH ||
    heightAfterRotation * numerator / JpegTranscoder.SCALE_DENOMINATOR > MAX_HEIGHT) {
  numerator--;
}
```